From 5e69bd4d4f5dab2780f2163d150994d293f5415e Mon Sep 17 00:00:00 2001
From: Liron Himi <lironh@marvell.com>
Date: Wed, 17 Apr 2019 19:26:28 +0300
Subject: [PATCH 153/386] net: mvpp2: reassociate pool to rxqs

In case interface is move from MUSDK to non-MUSDK mode, the rxqs
must be reassociate with its original pool's ids.
Without this patch the rxqs are left with the MUSDK's pools which
deny any new frames to received.

Change-Id: Id98d6a66b4c9acb2f004479f9bccb857d7293d5c
Signed-off-by: Liron Himi <lironh@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/7778
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Liron Himi <liron.himi@marvell.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 11b1cc17f3dd..c8a012a4aa43 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -5332,6 +5332,7 @@ static int mvpp2_port_musdk_cfg(struct net_device *dev, bool ena)
 	} *us;
 
 	struct mvpp2_port *port = netdev_priv(dev);
+	int rxq;
 
 	if (ena) {
 		/* Disable Queues and IntVec allocations for MUSDK,
-- 
2.17.1

